lib/libutee/tee_api_objects.c

```diff
@@ -614,8 +650,11 @@ TEE_Result TEE_GetNextPersistentObject(TEE_ObjectEnumHandle objectEnumerator,
 		goto out;
 	}
 
-	res = utee_storage_next_enum(objectEnumerator, objectInfo, objectID,
-				     objectIDLen);
+	len = *objectIDLen;
+	res = utee_storage_next_enum((uintptr_t)objectEnumerator, objectInfo,
+				     objectID, &len);
+	if (res == TEE_SUCCESS)
+		*objectIDLen = len;
```

Comment:
Would prefer to keep the previous behavior, that is update `*objectIDLen = len` what ever is the result, as this change is not 64bits related.